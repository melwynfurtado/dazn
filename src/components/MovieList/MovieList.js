import React from 'react'
import PropTypes from 'prop-types'
import Movie from './Movie'
import './MovieList.scss'

const MovieList =({ movies }) => (
  <ul className="search-results">
    {
      movies.length === 0 ? 
        <li>No movies were retrieved.</li>
      :
        movies.map(movie => <Movie key={movie.id} movie={movie} />)
    }
  </ul>
)

MovieList.propTypes = {
  movies: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,  
    })
  ),
}

export default MovieList