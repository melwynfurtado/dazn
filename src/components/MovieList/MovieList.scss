.search-results {
  text-align: left;
  margin: 20px;
  list-style: none;
}