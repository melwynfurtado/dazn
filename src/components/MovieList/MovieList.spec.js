import React from 'react'
import renderer from 'react-test-renderer'
import MovieList from './MovieList'

describe('MovieList Component', () => {
  const movies = [{
    id: 123,
    title: 'Rambo II',
  }]

  it('renders correctly', () => {
    const tree = renderer
                  .create(<MovieList movies={movies} />)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders "No results found"', () => {
    const tree = renderer
                  .create(<MovieList movies={[]} />)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })
})