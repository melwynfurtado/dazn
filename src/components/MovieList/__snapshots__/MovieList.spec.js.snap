// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList Component renders "No results found" 1`] = `
<ul
  className="search-results"
>
  <li>
    No movies were retrieved.
  </li>
</ul>
`;

exports[`MovieList Component renders correctly 1`] = `
<ul
  className="search-results"
>
  <li
    className="movie"
    onClick={[Function]}
  >
    Rambo II
  </li>
</ul>
`;
